### PR TITLE
Simple fixes to build on MacOSx

### PR DESCRIPTION
**Pull request by digantdesai**

### CODE REVIEW

File: third_party/cpu/triton_cpu.cc [hunk 1]
```diff
@@ -159,7 +161,7 @@ void init_triton_cpu(py::module &&m) {
   init_triton_cpu_passes_ttcpuir(passes.def_submodule("ttcpuir"));
 
   m.def("enable_amx", []() -> bool {
-#ifdef ARCH_REQ_XCOMP_PERM
+#if defined(__linux__) && defined(ARCH_REQ_XCOMP_PERM)
```
Comment (int3) [inline third_party/cpu/triton_cpu.cc:164]: I don't think the `defined(__linux__)` here is necessary?

**Review by int3 — APPROVED** [review]
lgtm

File: third_party/cpu/triton_cpu.cc (hunk 1 above)
Comment (digantdesai) [reply third_party/cpu/triton_cpu.cc:164]: Yeah, agree. I guess explicit is better?

**Comment by digantdesai** [comment]
Oh sorry about that. I saw the CI failure, but didn't realize that the CI failures were related to that.

**Comment by digantdesai** [comment]
`python: /home/runner/work/triton/triton/llvm-project/llvm/lib/IR/Value.cpp:333: void llvm::Value::setNameImpl(const llvm::Twine &): Assertion `!NameRef.contains(0) && "Null bytes are not allowed in names"' failed.`
Oh I see it now. Let me fix it.

**Pull request merged**